fe1-web/parts/Wallet.tsx

```diff
@@ -0,0 +1,185 @@
+              ? setWalletState(STRINGS.wallet_state_initialized)
+              : setWalletState(STRINGS.wallet_state_null)))}
+        />
+        <WideButtonView
+          title={STRINGS.back_to_wallet_home}
+          onPress={() => setWalletState(STRINGS.wallet_state_null)}
+        />
+      </View>
+    );
+  }
+
+  function getShowSeedWalletDisplay() {
+    return (
+      <View style={styleContainer.centered}>
+        <TextBlock bold text={STRINGS.show_seed_info} />
+        <TextBlock text={' '} />
+        <TextBlock text={seed} />
+        <TextBlock text={' '} />
+        <WideButtonView
+          title={STRINGS.back_to_wallet_home}
+          onPress={() => setWalletState(STRINGS.wallet_state_null)}
+        />
+      </View>
+    );
+  }
+
+  function showHideSeed() {
+    return (
+      <View>
+        <TextBlock visibility={visibleSeed} text={recoveredSeed} />
+        <TextBlock text={' '} />
+        <WideButtonView
+          title={showHideSeedTitle}
+          onPress={() => {
+            setShowHideSeedTitle(visibleSeed ? STRINGS.show_seed_title : STRINGS.hide_seed_title);
+            setVisibleSeed(!visibleSeed);
+          }}
+        />
+        <TextBlock text={' '} />
+        <TextBlock visibility={visibleEncryptedSeed} text={recoveredEncryptedSeed} />
+        <TextBlock text={' '} />
+        <WideButtonView
+          title={showHideEncryptedSeedTitle}
+          onPress={() => {
+            setShowHideEncryptedSeedTitle(visibleEncryptedSeed
+              ? STRINGS.show_encrypted_seed_title
+              : STRINGS.hide_encrypted_seed_title);
+            setVisibleEncryptedSeed(!visibleEncryptedSeed);
+          }}
+        />
+      </View>
+    );
+  }
+
+  function recoverTokens() {
+    return (
+      <View>
+        <WideButtonView
+          title={STRINGS.test_recover_seed_from_state}
+          onPress={() => {
+            setRecoveredEncryptedSeed(wallet.getEncryptedSeedToUint8Array().toString());
+            wallet.getDecryptedSeed().then((s) => setRecoveredSeed(s.toString()))
+              .catch((e) => console.log(e.toString()));
+            setTokensRecovered(true);
+          }}
+        />
+      </View>
+    );
+  }
+
+  function getWalletDisplay() {
+    return (
+      <View style={styleContainer.centered}>
+        <TextBlock bold text={STRINGS.wallet_synced_info} />
+        <TextBlock text={' '} />
+        { !tokensRecovered && recoverTokens() }
+        { tokensRecovered && showHideSeed() }
+      </View>
+    );
+  }
+
+  function getDisplay() {
+    return (
+      <View style={styleContainer.centered}>
+        { walletState === STRINGS.wallet_state_null && getStartWalletDisplay() }
+        { walletState === STRINGS.wallet_state_show_seed && getShowSeedWalletDisplay() }
+        { walletState === STRINGS.wallet_state_set_seed && getInsertSeedWalletDisplay() }
+        { walletState === STRINGS.wallet_state_initialized && getWalletDisplay() }
```

Comment:
if I understand correctly what is going on, you're rolling your own state machines to display 4 completely different screens within one part. 

It seems to me like the correct approach would be to use React's navigation capabilities in this case.